### ISSUE TITLE
old CLI `--dir` arg clobbers new CLI `--dir` argument

### ISSUE THREAD

**Issue by ydnar** [bytecodealliance/wasmtime#7415]
Thanks for releasing the compatibility shim in v14.0.3! I ran into an odd compatibility issue:

If `wasmtime` is run without `WASMTIME_NEW_CLI=1`, then `--dir` is always interpreted as a single value, and not `HOST::GUEST`:

```sh
wasmtime run --dir ./::./ ...
```

Fails with:

```sh
Error: failed to open directory './::./'
```

**Comment by alexcrichton** [comment]
Thanks for the report, I've opened https://github.com/bytecodealliance/wasmtime/pull/7416 to fix the warning here and we can do a 14.0.4 release as well in a day or two to collect some more feedback.

I'll also confirm though that Wasmtime doesn't just fail with the error message above, it also prints the expected warning about the old-vs-new CLI, which is what I at least expected. Still something that's worth fixing because otherwise `--dir`-with-remapping behavior isn't actually able to be used without a warning or env var, which isn't great.

**Comment by alexcrichton** [comment]
Fixed in https://github.com/bytecodealliance/wasmtime/pull/7416, backported in https://github.com/bytecodealliance/wasmtime/pull/7421, and 14.0.4 will go out when https://github.com/bytecodealliance/wasmtime/pull/7431 is merged, so closing. Thanks again for the report!
